Application fails to load on Linux

Using Ubuntu 20.04 LTS, OpenJDK 11.0.8:
```bash
wltan@GPBox:~/Downloads$ java -version
openjdk version "11.0.8" 2020-07-14
OpenJDK Runtime Environment (build 11.0.8+10-post-Ubuntu-0ubuntu120.04)
OpenJDK 64-Bit Server VM (build 11.0.8+10-post-Ubuntu-0ubuntu120.04, mixed mode, sharing)
```

I get the following error when launching the application by command line:
```bash
wltan@GPBox:~/Downloads$ java -jar finesse.jar
Oct 29, 2020 3:59:39 PM ay2021s1_cs2103_w16_3.finesse.MainApp init
INFO: =============================[ Initializing Fine$$e ]===========================
Oct 29, 2020 3:59:39 PM ay2021s1_cs2103_w16_3.finesse.MainApp initConfig
INFO: Using config file : config.json
Oct 29, 2020 3:59:39 PM ay2021s1_cs2103_w16_3.finesse.commons.util.JsonUtil readJsonFile
INFO: Json file config.json not found
Oct 29, 2020 3:59:39 PM ay2021s1_cs2103_w16_3.finesse.MainApp initPrefs
INFO: Using prefs file : preferences.json
Oct 29, 2020 3:59:39 PM ay2021s1_cs2103_w16_3.finesse.commons.util.JsonUtil readJsonFile
INFO: Json file preferences.json not found
Oct 29, 2020 3:59:39 PM ay2021s1_cs2103_w16_3.finesse.commons.core.LogsCenter init
INFO: currentLogLevel: INFO
Oct 29, 2020 3:59:39 PM ay2021s1_cs2103_w16_3.finesse.commons.util.JsonUtil readJsonFile
INFO: Json file data/fine$$e.json not found
Oct 29, 2020 3:59:39 PM ay2021s1_cs2103_w16_3.finesse.MainApp initModelManager
INFO: Data file not found. Will be starting with a sample FinanceTracker
Oct 29, 2020 3:59:39 PM ay2021s1_cs2103_w16_3.finesse.model.util.SampleDataUtil getSampleFinanceTracker
INFO: Populating Fine$$e with sample data...
Oct 29, 2020 3:59:39 PM ay2021s1_cs2103_w16_3.finesse.MainApp start
INFO: Starting Fine$$e V0.6.0ea
Oct 29, 2020 3:59:39 PM ay2021s1_cs2103_w16_3.finesse.ui.UiManager start
INFO: Starting UI...
Oct 29, 2020 3:59:40 PM ay2021s1_cs2103_w16_3.finesse.ui.UiManager start
SEVERE: null
java.lang.NullPointerException
	at ay2021s1_cs2103_w16_3.finesse.ui.MainWindow.setWindowDefaultSize(MainWindow.java:171)
	at ay2021s1_cs2103_w16_3.finesse.ui.MainWindow.<init>(MainWindow.java:93)
	at ay2021s1_cs2103_w16_3.finesse.ui.UiManager.start(UiManager.java:44)
	at ay2021s1_cs2103_w16_3.finesse.MainApp.start(MainApp.java:176)
	at com.sun.javafx.application.LauncherImpl.lambda$launchApplication1$9(LauncherImpl.java:846)
	at com.sun.javafx.application.PlatformImpl.lambda$runAndWait$12(PlatformImpl.java:455)
	at com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:428)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:427)
	at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:96)
	at com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	at com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$11(GtkApplication.java:277)
	at java.base/java.lang.Thread.run(Thread.java:834)

Oct 29, 2020 3:59:40 PM ay2021s1_cs2103_w16_3.finesse.ui.UiManager showFatalErrorDialogAndShutdown
SEVERE: Fatal error during initializing nullnull
java.lang.NullPointerException
	at ay2021s1_cs2103_w16_3.finesse.ui.MainWindow.setWindowDefaultSize(MainWindow.java:171)
	at ay2021s1_cs2103_w16_3.finesse.ui.MainWindow.<init>(MainWindow.java:93)
	at ay2021s1_cs2103_w16_3.finesse.ui.UiManager.start(UiManager.java:44)
	at ay2021s1_cs2103_w16_3.finesse.MainApp.start(MainApp.java:176)
	at com.sun.javafx.application.LauncherImpl.lambda$launchApplication1$9(LauncherImpl.java:846)
	at com.sun.javafx.application.PlatformImpl.lambda$runAndWait$12(PlatformImpl.java:455)
	at com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:428)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:427)
	at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:96)
	at com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	at com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$11(GtkApplication.java:277)
	at java.base/java.lang.Thread.run(Thread.java:834)

Exception in Application start method
Oct 29, 2020 3:59:40 PM ay2021s1_cs2103_w16_3.finesse.MainApp stop
INFO: ============================ [ Stopping Fine$$e ] =============================
Exception in thread "main" java.lang.RuntimeException: Exception in Application start method
	at com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:900)
	at com.sun.javafx.application.LauncherImpl.lambda$launchApplication$2(LauncherImpl.java:195)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.NullPointerException
	at ay2021s1_cs2103_w16_3.finesse.ui.UiManager.showAlertDialogAndWait(UiManager.java:64)
	at ay2021s1_cs2103_w16_3.finesse.ui.UiManager.showFatalErrorDialogAndShutdown(UiManager.java:89)
	at ay2021s1_cs2103_w16_3.finesse.ui.UiManager.start(UiManager.java:50)
	at ay2021s1_cs2103_w16_3.finesse.MainApp.start(MainApp.java:176)
	at com.sun.javafx.application.LauncherImpl.lambda$launchApplication1$9(LauncherImpl.java:846)
	at com.sun.javafx.application.PlatformImpl.lambda$runAndWait$12(PlatformImpl.java:455)
	at com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:428)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:427)
	at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:96)
	at com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	at com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$11(GtkApplication.java:277)
	... 1 more
```
Successive runs of the same command yield the same result.

On closer inspection, it looks like the default generated `preferences.json` contains `"windowCoordinates" : null` which might be the source of the problem.
```json
{
  "guiSettings" : {
    "windowWidth" : 1240.0,
    "windowHeight" : 815.0,
    "windowCoordinates" : null,
    "isFullscreen" : false
  },
  "financeTrackerFilePath" : "data/fine$$e.json"
}
```

--- a/src/main/java/ay2021s1_cs2103_w16_3/finesse/ui/MainWindow.java
+++ b/src/main/java/ay2021s1_cs2103_w16_3/finesse/ui/MainWindow.java
@@ -168,8 +168,10 @@ public class MainWindow extends UiPart<Stage> {
         // Retrieve window size and position from previous session.
         windowHeight = guiSettings.getWindowHeight();
         windowWidth = guiSettings.getWindowWidth();
-        xPosition = (int) guiSettings.getWindowCoordinates().getX();
-        yPosition = (int) guiSettings.getWindowCoordinates().getY();
+        if (guiSettings.getWindowCoordinates() != null) {
+            xPosition = (int) guiSettings.getWindowCoordinates().getX();
+            yPosition = (int) guiSettings.getWindowCoordinates().getY();
+        }
 
         // Restore window size and position from previous session.
         primaryStage.setHeight(windowHeight);

